src/main/scala/org/broadinstitute/dsde/agora/server/business/PermissionBusiness.scala

```diff
@@ -1,105 +1,167 @@
-    NamespacePermissionsClient.deleteNamespacePermission(entity, userToRemove)
+  def editNamespacePermission(entity: AgoraEntity, requester: String, accessObject: AccessControl): Future[Int] = {
+    permissionsDataSource.inTransaction { db =>
+      authNamespaceRequester(db, entity, requester, accessObject) {
+        db.nsPerms.editNamespacePermission(entity, accessObject)
+      }
+    }
   }
 
-  def listEntityPermissions(entity: AgoraEntity, requester: String): Seq[AccessControl] = {
-    authorizeEntityRequester(entity, requester)
-    AgoraEntityPermissionsClient.listEntityPermissions(entity)
+  def deleteNamespacePermission(entity: AgoraEntity, requester: String, userToRemove: String): Future[Int] = {
+    Try(checkSameRequester(requester, userToRemove)) match {
+      case Failure(regret) => Future.failed(regret)
+      case Success(_) => {
+        permissionsDataSource.inTransaction { db =>
+          authNamespaceRequester(db, entity, requester) {
+            db.nsPerms.deleteNamespacePermission(entity, userToRemove)
+          }
+        }
+      }
+    }
   }
 
-  def insertEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl): Int = {
-    authorizeEntityRequester(entity, requester, accessObject)
-    AgoraEntityPermissionsClient.insertEntityPermission(entity, accessObject)
+  def listEntityPermissions(entity: AgoraEntity, requester: String): Future[Seq[AccessControl]] = {
+    permissionsDataSource.inTransaction { db =>
+      authEntityRequester(db, entity, requester) {
+        db.aePerms.listEntityPermissions(entity)
+      }
+    }
   }
 
-  def batchEntityPermission(entity: AgoraEntity, requester: String, accessObjectList: List[AccessControl]): Int = {
-    // Batch authorize, then batch insert.
-    accessObjectList.foreach(accessObject => authorizeEntityRequester(entity, requester, accessObject))
-    accessObjectList.map(accessObject => AgoraEntityPermissionsClient.insertEntityPermission(entity, accessObject)).sum
+  def insertEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl): Future[Int] = {
+    permissionsDataSource.inTransaction { db =>
+      authEntityRequester(db, entity, requester, accessObject) {
+        db.aePerms.insertEntityPermission(entity, accessObject)
+      }
+    }
   }
 
-  def editEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl): Int = {
-    authorizeEntityRequester(entity, requester, accessObject)
-    AgoraEntityPermissionsClient.editEntityPermission(entity, accessObject)
+  def batchEntityPermission(entity: AgoraEntity, requester: String, accessObjectList: List[AccessControl]): Future[Int] = {
+    def authOne(db: DataAccess, accessObject: AccessControl) = {
+      authEntityRequester(db, entity, requester, accessObject) {
+        DBIOAction.successful()
+      }
+    }
+    permissionsDataSource.inTransaction { db =>
+      // Batch authorize, then batch insert.
```

Comment:
It's still all done in one same transaction. It's likely one could write a batch-insert function for both of these to cut down on the number of SQL queries, but I'd consider that an enhancement :)